### PR TITLE
[CUBLAS][TF32][CUDNN] Update numerical_accuracy.rst

### PR DESCRIPTION
**Pull request by eqy**
CC @mruberry @ptrblck

### CODE REVIEW

**Comment by eqy** [comment]
Cherry-pick PR: #79539

File: docs/source/notes/numerical_accuracy.rst [hunk 1]
```diff
@@ -54,16 +54,14 @@ datatype. E.g.:
...
-results, and surprising results such as multiplying a matrix by identity matrix produces
-results that are different from the input.
-Most neural network workloads have the same convergence behavior when using tf32 as they have
-with fp32, however, if better accuracy is desired, TF32 can be turned off with
-``torch.backends.cuda.matmul.allow_tf32 = False``
-
-For more information see :ref:`TensorFloat32<tf32_on_ampere>`
+On Ampere Nvidia GPUs, PyTorch can use TensorFloat32 (TF32) to speed up mathematically intensive operations, in particular matrix multiplications and convolutions.
+When an operation is performed using TF32 tensor cores, only the first 10 bits of the input mantissa are read.
+This elision leads to less accurate results, and surprising results (e.g., multiplying a matrix by the identity matrix produces results that are different from the input).
```
Comment (mruberry) [inline docs/source/notes/numerical_accuracy.rst:59]: "This may reduce accuracy and produce surprising results ..."

Comment (mruberry) [reply docs/source/notes/numerical_accuracy.rst:59]: "... by the identity matrix may ..."

File: docs/source/notes/numerical_accuracy.rst [hunk 2]
```diff
@@ -54,16 +54,14 @@ datatype. E.g.:
...
+By default, this option is disabled for matrix multiplications and enabled for convolutions, although most neural network workloads have the same convergence behavior when using TF32 as they have with fp32.
```
Comment (mruberry) [inline docs/source/notes/numerical_accuracy.rst:60]: "this option" -> "using TF32 tensor cores"

File: docs/source/notes/numerical_accuracy.rst [hunk 3]
```diff
@@ -54,16 +54,14 @@ datatype. E.g.:
...
+However, if better throughput is desired for matrix multiplications, TF32 can be turned on with ``torch.backends.cuda.matmul.allow_tf32 = True``.
```
Comment (mruberry) [inline docs/source/notes/numerical_accuracy.rst:61]: "However, ..." -> "We recommend enabling TF32 tensor cores for matrix multiplications if your network does not need full float32 precision."

File: docs/source/notes/numerical_accuracy.rst [hunk 4]
```diff
@@ -54,16 +54,14 @@ datatype. E.g.:
...
+Conversely, if better accuracy is desired for convolutions, TF32 can be turned off with ``torch.backends.cudnn.allow_tf32 = False``.
```
Comment (mruberry) [inline docs/source/notes/numerical_accuracy.rst:62]: If your network needs full float32 precision for both matrix multiplications and convolutions, then TF32 tensor cores can also be disabled for convolutions using..."

**Review by mruberry — COMMENTED** [review]
Overall looks good, @eqy, I made a couple wording suggestions for your review

**Comment by eqy** [comment]
Thanks, should I open another cherry-pick PR with additional changes for the release branch?

**Comment by mruberry** [comment]
> Thanks, should I open another cherry-pick PR with additional changes for the release branch?

Release branch is finalized so don't worry about it

**Review by ngimel — APPROVED** [review]

**Review by mruberry — APPROVED** [review]
Looks great, thanks @eqy!

**Pull request closed without merging**